### PR TITLE
Fix: solution to double selection

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Currently there is cases where the keyboard interaction triggereds a selection but also triggers the click which then removed the selection.
This is begin fixed by this PR. which as well adds tab index for selectable components.

fixes https://github.com/umbraco/Umbraco.UI/issues/654

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

- [ ] If my change requires a change to the documentation, I have updated the documentation in this pull request.
- [ ] I have read the **[CONTRIBUTING](<(https://github.com/umbraco/Umbraco.UI/blob/v1/contrib/docs/CONTRIBUTING.md)>)** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/uui-base/lib/mixins/SelectableMixin.ts [hunk 1]
```diff
@@ -76,14 +76,20 @@ export const SelectableMixin = <T extends Constructor<LitElement>>(
     constructor(...args: any[]) {
       super(...args);
       this.addEventListener('click', this._handleClick);
-      this.addEventListener('keydown', this.handleSelectKeydown);
+      this.addEventListener('keypress', this.handleSelectKeydown);
```
Comment (bjarnef) [inline packages/uui-base/lib/mixins/SelectableMixin.ts:79]: We should use `keypress` event - it has been deprecated for a while know:
https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event

Comment (nielslyngsoe) [reply packages/uui-base/lib/mixins/SelectableMixin.ts]: Well spotted, thanks. We need to correct this across.

Comment (nielslyngsoe) [reply packages/uui-base/lib/mixins/SelectableMixin.ts]: And a PR for that: https://github.com/umbraco/Umbraco.UI/pull/960

File: packages/uui-base/lib/mixins/SelectableMixin.ts [hunk 2]
```diff
@@ -80,10 +80,16 @@ export const SelectableMixin = <T extends Constructor<LitElement>>(
...
-      if (this.selectableTarget === this) {
-        if (e.key !== ' ' && e.key !== 'Enter') return;
-        this._toggleSelect();
+      const composePath = e.composedPath();
+      if (
+        (this._selectable || (this.deselectable && this.selected)) &&
+        composePath.indexOf(this.selectableTarget) === 0
+      ) {
+        if (this.selectableTarget === this) {
+          if (e.key !== ' ' && e.key !== 'Enter') return;
```
Comment (madsrasmussen) [inline packages/uui-base/lib/mixins/SelectableMixin.ts:89]: Can we use e.code !== 'Space' here instead? I think that is safer than the empty space.

Comment (nielslyngsoe) [reply packages/uui-base/lib/mixins/SelectableMixin.ts]: good input

**Review by madsrasmussen — APPROVED** [review]

**Pull request merged**